### DISCUSSION TITLE
Local build

### DISCUSSION THREAD

**Discussion by epgfm** [FossifyOrg/General-Discussion#232] [General]
Hello
I'm trying to build some of these repositories locally but I'm having a problem with the gradle build.
It runs fine when the dependencies { implementation() } refers to the jitpack commons, but I can't get it to refer to the local version of commons I have previously built.
Can anyone point me in the right direction?

**Comment by naveensingh** [comment]
You can check settings.gradle.kts in Fossify Gallery. You will find an example there on how to include commons locally. Can't help more as I'm not on my computer.

**Comment by epgfm** [comment]
Hey @naveensingh 
Thanks for the reply
So I have tried to follow the example, my settings.gradle.kts file in Phone repository now looks like this:
```kts
rootProject.name = "Phone"
enableFeaturePreview("TYPESAFE_PROJECT_ACCESSORS")
pluginManagement {
    repositories {
        gradlePluginPortal()
        google()
        mavenCentral()
    }
}
dependencyResolutionManagement {
    repositoriesMode.set(RepositoriesMode.FAIL_ON_PROJECT_REPOS)
    repositories {
        google()
        mavenCentral()
        maven { setUrl("https://jitpack.io") }
    }
}
include(":app")
includeBuild("../Commons") {
    dependencySubstitution {
        substitute(module("org.fossify:commons")).using(project(":commons"))
    }
}
```
Doesn't work though, it looks like it's finding the path while building and I see it build the content, but after a while it fails with that error:
![image](https://github.com/user-attachments/assets/3b0b2c3f-b293-49a5-9125-74c47fd0ae1d)
I've been scratching my head for a while because of this. Can't get it to work.

**Reply by naveensingh** [reply]
As a workaround, you can comment out this line: 

https://github.com/FossifyOrg/Phone/blob/c15698d13e1b6703e748c16cfa7ad703c66e576b/app/src/main/kotlin/org/fossify/phone/dialogs/ChangeSortingDialog.kt#L42

```kotlin
// divider.beGoneIf(isCustomSorting)
```

Or replace it with the following:

```kotlin
divider.root.beGoneIf(isCustomSorting)
```
